src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -126,7 +140,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             Dim rightOperatorBinary As BoundUserDefinedBinaryOperator = TryCast(Me.Right, BoundUserDefinedBinaryOperator)
             If rightOperatorBinary IsNot Nothing Then
-                If TypeOf rightOperatorBinary.Left Is BoundCompoundAssignmentTargetPlaceholder Then
+                If TypeOf DirectCast(rightOperatorBinary, IBinaryOperatorExpression).LeftOperand Is BoundCompoundAssignmentTargetPlaceholder Then
```

Comment:
Agreed. The goal for Update 2 is to avoid crashes, and to handle simple cases correctly. For Update 3 this needs to be completely correct.